sygnal/gcmpushkin.py

```diff
@@ -354,7 +354,14 @@ async def _dispatch_notification_unlimited(
 
             # Reject pushkey if default_payload is misconfigured
             if data is None:
+                logger.warning(
+                    "Rejecting pushkey due to misconfigured default_payload, "
+                    "please ensure that default_payload is a dict."
+                )
                 failed.append(device.pushkey)
+                pushkeys.remove(device.pushkey)
+                if len(pushkeys) == 0:
+                    return failed
```

Comment:
We're about to use the same `data` for all the pushkeys in `pushkeys`, so I think we have to reject all of them instead of just one?

Reply:
Right that makes sense.